pkg/statedb/db.go

```diff
@@ -50,38 +51,38 @@ var _ DB = &stateDB{}
 // WriteJSON marshals out the whole database as JSON into the given writer.
 func (db *stateDB) WriteJSON(w io.Writer) error {
 	tx := db.memDB.Txn(false)
-	if _, err := w.Write([]byte("{\n")); err != nil {
-		return err
-	}
+	buf := bufio.NewWriter(w)
+	buf.WriteString("{\n")
+	first := true
 	for table := range db.memDB.DBSchema().Tables {
+		if !first {
+			buf.WriteString(",\n")
+		} else {
+			first = false
+		}
 		iter, err := tx.Get(table, "id")
 		if err != nil {
 			return err
 		}
-		if _, err := w.Write([]byte("\"" + table + "\": [\n")); err != nil {
-			return err
-		}
+		buf.WriteString("\"" + table + "\": [\n")
```

Comment:
Bit weird to ignore errors in some places but not in other places - either throw it everywhere or ignore it, I guess?

Reply:
The buffered writer carries the error over multiple calls and I only need to care about the error on `Flush()`. Removed the unnecessary checks.

Reply:
oh yeah, good to know - I guess it's not really part of the documented API but I doubt that it'll ever change.